tests/Integration/features/bootstrap/FeatureContext.php

```diff
@@ -964,21 +1040,39 @@ public function userDeletesPublicShare(string $user, string $collective): void {
 	public function anonymousSeesPublicCollective(string $collective, string $owner): void {
 		$this->setCurrentUser($owner);
 		$collectiveId = $this->collectiveIdByName($collective);
-		$token = $this->getCollectiveShareToken($collectiveId);
+		$token = $this->getShareToken($collectiveId);
 		$this->sendRequest('GET', '/apps/collectives/_api/p/' . $token, null, null, [], false);
 		$this->assertStatusCode(200);
 		$this->assertCollectiveByName($collective);
 		$this->assertCollectiveLevel($collective, 1);
 	}
 
 	/**
-	 * @When anonymous fails to see public collective :collective with stored token
+	 * @When anonymous sees public page share :page in collective :collective with owner :owner
 	 *
+	 * @param string $page
 	 * @param string $collective
+	 * @param string $owner
 	 *
 	 * @throws GuzzleException
 	 */
-	public function anonymousFailsToSeePublicCollective(string $collective): void {
+	public function anonymousSeesPublicPageShare(string $page, string $collective, string $owner): void {
+		$this->setCurrentUser($owner);
```

Comment:
I don't quite understand what `setCurrentUser` does. But my intuition says for testing anonymous behavior it should not be set to the owner.

Is the current user only needed for the preparation and `sendRequest` then does not use it anymore?

Reply:
Exactly. `setCurrentUser` is needed to retrieve the `pageId`, as we don't know that one. The request itself doesn't use it, because `sendRequest` get's `auth: false` passed.